### PR TITLE
[SYCL][NFC] avoid redundant copies of platform/device vector 

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

File: sycl/tools/sycl-ls/sycl-ls.cpp [hunk 1]
```diff
@@ -106,7 +106,7 @@ int main(int argc, char **argv) {
     return EXIT_FAILURE;
   }
 
-  auto Platforms = platform::get_platforms();
+  const auto &Platforms = platform::get_platforms();
```
Comment (romanovvlad) [inline sycl/tools/sycl-ls/sycl-ls.cpp:109]: Could you please elaborate more on how the change ensures that lifetime is correct? What's wrong with current code?

Comment (smaslov-intel) [reply sycl/tools/sycl-ls/sycl-ls.cpp:109]: It makes the vector of platforms/devices returned by the get_platfroms/get_devices live until the reference to them is out of scope. Currently a copy of vector was made, and the original returned vector destructed.

Comment (romanovvlad) [reply sycl/tools/sycl-ls/sycl-ls.cpp:109]: Since sycl classes are reference countable, lifetime is the same in both cases.

Comment (smaslov-intel) [reply sycl/tools/sycl-ls/sycl-ls.cpp:109]: OK, then this must be NFC, and I reflected this in the title.

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**